l10n_nl_partner_name/models/res_partner.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2017-24 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    """Extend res.partner with extra fields for Dutch names."""
+
+    _inherit = "res.partner"
+
+    initials = fields.Char()
+    infix = fields.Char()
+
+    @api.depends("firstname", "lastname", "initials", "infix")
+    def _compute_name(self):
+        for record in self:
+            record.name = record._get_computed_name(
+                record.lastname,
+                record.firstname,
+                initials=record.initials,
+                infix=record.infix,
+            )
+
+    @api.model
+    def _get_computed_name(self, lastname, firstname, initials=None, infix=None):
+        return " ".join(p for p in (firstname, initials, infix, lastname) if p)
```

Comment:
This is to simple as it does not take into account the configured setiing for name order in partner_firstname.

The proper way I think would be to compute the firstname to use in super, than call super:
```
dutch_firstname = " ".join(p for p in (firstname, initials, infix) if p)
return super()._get_computed_name(dutch_firstname, lastname)
```

The data with default can be deleted, as firstname_lastname is the default in partner_firstname anyway.